### PR TITLE
Update Code to 1.90.0

### PR DESCRIPTION
**Pull request by benz0li**
- Update Node to 20.11.1
- Update documentation

Fixes https://github.com/coder/code-server/issues/6822

### CODE REVIEW

**Comment by benz0li** [comment]
`code-server-4.90.0-rc.1-linux-amd64.tar.gz` with Code 1.90.0 is deployed at https://coder.jupyter.b-data.ch.

Functionality [modified by patches] tested and fond **not** to work:

* marketplace

Neither working in GUI nor Terminal. Output when executing e.g. `code-server --install-extension alefragnani.project-manager`

```
#8 11.48 Error while installing extensions: End of central directory record signature not found. Either not a zip file, or file is truncated.
#8 11.48 End of central directory record signature not found. Either not a zip file, or file is truncated.
#8 11.48 [2024-06-06T08:35:55.162Z] error parent:70 Uncaught exception: End of central directory record signature not found. Either not a zip file, or file is truncated.
#8 11.48 [2024-06-06T08:35:55.162Z] error parent:70 Error: End of central directory record signature not found. Either not a zip file, or file is truncated.
#8 11.48     at y (/opt/code-server/lib/vscode/out/vs/server/node/server.main.js:105:2374)
#8 11.48     at /opt/code-server/lib/vscode/out/vs/server/node/server.main.js:105:3812
#8 11.48     at /opt/code-server/lib/vscode/node_modules/yauzl/index.js:40:7
#8 11.48     at /opt/code-server/lib/vscode/node_modules/yauzl/index.js:190:5
#8 11.48     at /opt/code-server/lib/vscode/node_modules/yauzl/index.js:705:5
#8 11.48     at /opt/code-server/lib/vscode/node_modules/yauzl/fd-slicer.js:33:7
#8 11.48     at callback (/opt/code-server/lib/vscode/node_modules/graceful-fs/polyfills.js:138:21)
#8 11.48     at FSReqCallback.wrapper [as oncomplete] (node:fs:682:5)
```

**Comment by benz0li** [comment]
Functionality [modified by patches] tested and found to work:

* base-path
* cli-window-open
* local-storage
* proxy-url
* service-worker
* web view

Workspaces and Jupyter Notebooks also work fine:

*  `ms-toolsai.jupyter@2024.5.0`

ℹ️ Installing extensions from VSIX works.

**Comment by benz0li** [comment]
I followed the same procedure as with all previous release candidates. No idea what I might have missed.

@code-asher Please review.

File: patches/base-path.diff [hunk 1]
```diff
@@ -140,11 +140,11 @@ Index: code-server/lib/vscode/src/vs/server/node/webClientServer.ts
 +		const base = relativeRoot(getOriginalUrl(req))
 +		const vscodeBase = relativePath(getOriginalUrl(req))
 +
- 		const productConfiguration = <Partial<IProductConfiguration>>{
+ 		const productConfiguration = {
```
Comment (benz0li) [inline patches/base-path.diff:143]: This...

File: patches/marketplace.diff [hunk 2]
```diff
@@ -53,16 +53,16 @@ Index: code-server/lib/vscode/src/vs/server/node/webClientServer.ts
...
--				'resourceUrlTemplate': this._webExtensionResourceUrlTemplate.with({
+-				resourceUrlTemplate: this._webExtensionResourceUrlTemplate.with({
 -					scheme: 'http',
 -					authority: remoteAuthority,
 -					path: `${this._webExtensionRoute}/${this._webExtensionResourceUrlTemplate.authority}${this._webExtensionResourceUrlTemplate.path}`
 -				}).toString(true)
 -			} : undefined
 +			extensionsGallery: this._productService.extensionsGallery,
- 		};
+ 		} satisfies Partial<IProductConfiguration>;
```
Comment (benz0li) [inline patches/marketplace.diff:65]: ...moved to here.

**Comment by code-asher** [comment]
I think the marketplace error is related to Microsoft.VisualStudio.Services.VsixSignature.  From Microsoft's repository, this is a zip file, but from Open VSX it is not.  I will do some more investigation, but maybe we will just have to disable signature verification.

**Comment by code-asher** [comment]
Ah yeah here is the issue https://github.com/eclipse/openvsx/issues/922

**Review by code-asher — APPROVED** [review]
Thank you for putting this together!  I will put out an RC soon.

**Comment by benz0li** [comment]
I would like to confirm, that the marketplace is working fine with code-server [v4.90.0-rc.1](https://github.com/coder/code-server/releases/tag/v4.90.0-rc.1)

Python extension, too: `ms-python.python@2024.4.1`

**Comment by code-asher** [comment]
Awesome thank you for confirming.

I am seeing a weird thing where the first web socket connection keeps timing out (seeing `the handshake timed out` in the browser console), making it appear like loading code-server takes a really long time.  I am trying to figure out what it is, but I was curious to see if you were experiencing the same?

**Comment by benz0li** [comment]
> but it is timing out spawning at the moment.

Just try again. It will work eventually.

**Comment by benz0li** [comment]
> but I was curious to see if you were experiencing the same?

No, I do not experience this on https://coder.jupyter.b-data.ch.

**Comment by code-asher** [comment]
Very weird, I tried clearing the CI cache to force a rebuild of VS Code and for some reason CI keeps dumping core, so I am trying 20.12.1 instead.  It looks like VS Code uses the .nvmrc (20.12.1) to build, but then they ship the web version with the .yarnrc (20.11.1).

I have no idea why they use two different versions...I am going to just try building a new RC 20.12.1 and hopefully that works fine.

**Comment by benz0li** [comment]
> > but I was curious to see if you were experiencing the same?
> 
> No, I do not experience this on https://coder.jupyter.b-data.ch.

Did https://github.com/coder/code-server/commit/a73549539bb205169189158532c32d0fc44b104e resolve the web socket connection time outs?

ℹ️ In b-data's/my docker images `--disable-telemetry` is used when `code-server` is started.

**Comment by code-asher** [comment]
> Did https://github.com/coder/code-server/commit/a73549539bb205169189158532c32d0fc44b104e resolve the web socket connection time outs?

It did, yup!  In our environment for some reason `os.cpus()` is very slow (1-2 seconds), maybe because there are 64 cores?  On my laptop it takes around 0.04 seconds.   I think probably what happened is VS Code added a bunch of new telemetry calls on startup, and those new calls caused enough delay due to the slow `os.cpus()` to trigger the timeout.

> In b-data's/my docker images --disable-telemetry is used when code-server is started.

Ahhh that makes sense.

**Pull request merged**